src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -292,6 +292,38 @@ def individual_node_complexity(self) -> Dict[PlanNodeCategory, int]:
         )
 
 
+class UnresolvedColumnRegex(Expression):
+    def __init__(self, expressions: List[Attribute]) -> None:
+        super().__init__()
+        self.expressions = expressions
+
+    def dependent_column_names(self) -> Optional[AbstractSet[str]]:
+        # When the column is `df['*']`, `expressions` contains Attributes from all columns
+        # When the column is `col('*')` or just '*' string, `expressions` is empty,
+        # but its dependent columns should be all columns too
+        return (
+            derive_dependent_columns(*self.expressions)
+            if self.expressions
+            else COLUMN_DEPENDENCY_ALL
+        )
```

Comment:
I don't think this is correct. If we know expressions is a non-empty list, then
```suggestion
        return derive_dependent_columns(*self.expressions)
```